### ISSUE TITLE
Compile a single Whisper final decoder-init model for log probing

### ISSUE THREAD

**Issue by florianvoss-commit** [sima-neat/llima#133]
## Summary

Simplify Whisper log-probe compilation and runtime loading by compiling only one final decoder-init model. Keep the normal `decoder_init_layer<N>` name and conditionally include the logits output when log probing is enabled.

## Current behavior

When `log_probe_enabled` is enabled for Whisper Small, compilation produces both:

```text
models--openai--whisper-small_decoder_init_layer11_stage1_mla.elf
models--openai--whisper-small_decoder_init_log_probe_layer11_stage1_mla.elf
```

The regular init models are always generated for every decoder layer. Enabling log probe additionally generates a second variant for the final decoder layer.

The runtime likewise loads both final-layer models, but executes only one:

- `log_probe_enabled == false`: execute `decoder_init_layer11`.
- `log_probe_enabled == true`: execute `decoder_init_log_probe_layer11`.

Therefore, a package permanently compiled with log probing enabled contains and loads a regular layer-11 init model that is never executed.

The single-token decoder post path already follows the simpler behavior: `decoder_n1_post_layer11` keeps its normal name and conditionally exposes the additional logits output.

## Proposed behavior

Always compile exactly one final decoder-init model using the normal name:

```text
decoder_init_layer<last>
```

Configure its output ABI at compile time:

- Log probe disabled: output the selected token and cache tensors.
- Log probe enabled: additionally output the full filtered logits tensor.

The runtime should bind `decoder_logits` conditionally according to `log_probe_enabled` and always queue the regular decoder-init map for the final layer.

## Suggested cleanup

- Pass `enable_log_probe` to the normal final-layer `WhisperDecoderInitModel` when constructing the `init` model.
- Stop generating the separate `init_log_probe` model and ELF.
- Remove `_decoder_init_log_probe_model_map`.
- Remove the separate log-probe ELF path helper and runtime selection branch.
- Remove or deprecate the `init_log_probe` compilation part.
- Keep the existing `decoder_init_layer<N>` name for both configurations.

## Compatibility and artifact handling

This changes the output ABI of the final `decoder_init_layer<N>` ELF when log probing is enabled, so compiler and C++ runtime changes must land together.

Changing `log_probe_enabled` should require recompilation. Reusing an output directory with `--resume` must not silently retain a final-layer ELF generated with the opposite setting; either document that a clean rebuild is required or ensure stale-artifact detection covers this configuration change.

Existing deployed packages with `decoder_init_log_probe_layer<N>` may require a temporary runtime fallback if backward compatibility is required.

## Acceptance criteria

- A Whisper build contains only one final-layer decoder-init ELF.
- With log probing disabled, its outputs and runtime behavior remain unchanged.
- With log probing enabled, `decoder_init_layer<last>` emits filtered logits in addition to the selected token and caches.
- The runtime computes the same `avg_logprob` as before.
- No `decoder_init_log_probe_layer<N>` ELF is generated for new builds.
- Earlier decoder init layers, encoder, language detection, decoder pre/cache, and post behavior remain unchanged.
- Tests cover final-layer output bindings with log probing both enabled and disabled.
